lucet-module/src/module_data.rs

```diff
@@ -59,6 +59,8 @@ pub struct ModuleFeatures {
     pub lzcnt: bool,
     pub popcnt: bool,
     pub instruction_count: bool,
+    pub pinned_heap: bool,
```

Comment:
How's it sound to make this an `Option<HeapPinStyle>`? Then `None` means "no pinning", and `HeapPinStyle` can be written like
```
#[repr(u8)]
enum HeapPinStyle {
    PinR15 = 1
}
```
that we can extend if need be in the future?

I mostly want to avoid a cranelift `RegUnit`, even in `u16` form in `lucet-module` - the stability of those numbers isn't guaranteed, so we could misinterpret an old module, or an old runtime could misinterpret a new module.